### ISSUE TITLE
Missed a guard on hf_quantizer.check_quantized_param

### ISSUE THREAD

**Issue by pharaouk** [huggingface/transformers#28831]
### System Info

```
    model = AutoModelForCausalLM.from_pretrained(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/models/auto/auto_factory.py", line 562, in from_pretrained
    return model_class.from_pretrained(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/modeling_utils.py", line 3504, in from_pretrained
    ) = cls._load_pretrained_model(
  File "/root/miniconda3/envs/py3.10/lib/python3.10/site-packages/transformers/modeling_utils.py", line 3732, in _load_pretrained_model
    ) or not hf_quantizer.check_quantized_param(
AttributeError: 'NoneType' object has no attribute 'check_quantized_param'
```

only happens w multi gpu for some reason

missed a guard on hf_quantizers

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Private code

### Expected behavior

Code not breaking

**Comment by ArthurZucker** [comment]
cc @younesbelkada

**Comment by younesbelkada** [comment]
Yes, should be fixed by #28804
